hera_sim/cli.py

```diff
@@ -70,28 +85,35 @@ def run(input, outfile, verbose):
 
     sim = hera_sim.Simulator(**instrument_params)
 
+    # XXX: this section will take careful thought to implement correctly
     config_params = {}
-    # need to figure out a mapping of this to how defaults works
-    for component in ("systematics", "analysis",):
-        for key, value in yaml_contents[component].items():
-            config_params[key] = value
-    # this currently isn't ideal. there should be checks to see if there
-    # is any overlap between defaults and config_params; if there is
-    # an overlap, a warning should be raised. if there is no overlap,
-    # then the defaults and config_params dictionaries should be merged
-    # and the resulting dictionary should be applied as the new defaults
-    if not defaults and config_params:
-        hera_sim.defaults.set(**config_params)
-
+    # the following choice of syntax was scrapped and will be temporarily
+    # commented out before being removed entirely
+    #for component in ("analysis", "systematics",):
+    #    for key, value in yaml_contents[component].items():
+    #        config_params[key] = value
+    # default behavior should be such that anything specified in the
+    # configuration parameters will override whatever the current default
+    # settings are, but perhaps warn the user that something is being
+    # specified multiple times if defaults have been set and some of the
+    # defaults conflict with the configuration settings
+    # in the end, we want to handle the configuration stuff with the
+    # functionality offered by defaults
+    hera_sim.defaults.set(**config_params)
+
+    # extract the simulation parameters from the configuration file
+    # the configuration file should only specify any particular parameter once
+    # i.e. the same sky temperature model should be used throughout
     sim_params = {}
-    for key, value in yaml_contents["simulation"].items():
-        sim_params[key] = value
+    for component in yaml_contents["simulation"]["include"]:
+        for content in yaml_contents.values():
+            if component in content.keys():
+                for particular_component, parameters in content[component].items():
+                    sim_params[particular_component] = parameters
```

Comment:
I'm probably just missing it, but it's not clear to me how this will ensure that each parameter only needs to be specified once? It looks like it will have to be specified for each kind of component that's in `simulation['include']`?

Reply:
I'm pretty sure I was thinking that the `defaults` would take care of this, and that the contents in, say `yaml_contents['sky']['foregrounds']['diffuse_foreground']` would just specify the kwargs that are particular to that method